lib/beaker-hostgenerator/parser.rb

```diff
@@ -225,6 +235,11 @@ def settings_string_to_map(host_settings)
           raise BeakerHostGenerator::Exceptions::InvalidNodeSpecError,
                 "Malformed host settings: #{host_settings}"
         end
+        if pair[1].include?("[")
+          pair[1] = pair[1].
+          delete('[]').
+          split('~')
+        end
```

Comment:
I'd pull this out of the validation loop and give it its own loop with a comment.

Also do we need to validate the elements of the list in any way?  Not sure, maybe not.